### PR TITLE
Fix helm chart networking api version

### PR DESCRIPTION
**Pull request by gabrielemontano**
Fix networkingApi version in ingress manifest.
Add support for prometheus service monitor (with secret for basic authentication)

### CODE REVIEW

**Comment by mindrunner** [comment]
Any reason why this is not getting merged. I think prometheus support is very important.

**Comment by zmstone** [comment]
It was due to a wrong merge base.
The I guess it’s forgotten.

@gabrielemontano could you kindly rebase to main-v4.3 branch.

@Rory-Z could you see if we can pull the commit and create a new PR instead?

File: deploy/charts/emqx/templates/service-monitor.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+{{- if and (.Values.metrics.enabled) (eq .Values.metrics.type "prometheus") }}
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
```
[the remark this answers is not in the document]
Comment (mindrunner) [reply deploy/charts/emqx/templates/service-monitor.yaml:3]: This is pretty much standard in helm charts. I am not aware of noteworthy alternatives to prometheus. It is important, however, that this is an optional (which in this case is [`.Values.metrics.enabled`]) to maintain compatibility for clusters w/o prom-stack installed.

File: deploy/charts/emqx/templates/secret.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+{{- if .Values.metrics.enabled }}
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ include "emqx.fullname" . }}-basic-auth
+  namespace: {{ .Release.Namespace }}
+type: kubernetes.io/basic-auth
+stringData:
+  username: admin
+  password: public
```
Comment (Rory-Z) [inline deploy/charts/emqx/templates/secret.yaml:10]: Maybe we should configure it in `Value.yaml·, as far as I know, users can set their username and password by modifying emqx's environment variables

File: deploy/charts/emqx/templates/service-monitor.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  {{- if .Values.service.annotations }}
+  annotations:
+    {{ toYaml .Values.service.annotations | indent 4 }}
+  {{- end }} 
+spec:
+  endpoints:
+  - interval: 10s
+    port: mgmt 
+    scheme: http
+    path: /api/v4/emqx_prometheus
```
Comment (Rory-Z) [inline deploy/charts/emqx/templates/service-monitor.yaml:21]: This relies on the `emqx_prometheus` plugin for emqx, which we should probably turn on when we enabled metrics

File: deploy/charts/emqx/templates/secret.yaml (hunk 2 above)
Comment (mindrunner) [reply deploy/charts/emqx/templates/secret.yaml:10]: prometheus operator needs both values to be present in a secret. (not a configmap) [AFAIK]

**Comment by mindrunner** [comment]
What do we need to get this moving soon? Would it help to separate the patches from each other to have a dedicated monitoring PR?

**Comment by Rory-Z** [comment]
> What do we need to get this moving soon? Would it help to separate the patches from each other to have a dedicated monitoring PR?

Please check https://github.com/emqx/emqx/pull/7643

**Comment by zmstone** [comment]
This pr has a wrong merge base. 
if it’s intended for 5.0 #7643 has is covered.
If it’s for 4.3, please rebase.

**Comment by id** [comment]
It's been too long, closing.

**Pull request closed without merging**